plugins/applyTransformsShapes.js

```diff
@@ -0,0 +1,233 @@
+          ry = Number.isNaN(ry) ? rx || 0 : ry;
+
+          const cornerA = transformAbsolutePoint(matrix.data, x, y);
+          const cornerB = transformAbsolutePoint(
+            matrix.data,
+            x + width,
+            y + height,
+          );
+          const cornerX = Math.min(cornerA[0], cornerB[0]);
+          const cornerY = Math.min(cornerA[1], cornerB[1]);
+          const [newW, newH] = transformSize(matrix.data, width, height);
+          const [newRx, newRy] = transformSize(matrix.data, rx, ry);
+
+          if (strokeWidth) {
+            node.attributes['stroke-width'] = stringifyNumber(
+              strokeWidth * scale,
+              factor,
+              leadingZero,
+            );
+          }
+          node.attributes.x = stringifyNumber(cornerX, factor, leadingZero);
+          node.attributes.y = stringifyNumber(cornerY, factor, leadingZero);
+          node.attributes.width = stringifyNumber(newW, factor, leadingZero);
+          node.attributes.height = stringifyNumber(newH, factor, leadingZero);
+          if (newRx < 1 / factor && newRy < 1 / factor) {
+            delete node.attributes.rx;
+            delete node.attributes.ry;
+          } else if (Math.abs(newRx - newRy) < 1 / factor) {
+            node.attributes.rx = stringifyNumber(newRx, factor, leadingZero);
+            delete node.attributes.ry;
+          } else {
+            node.attributes.rx = stringifyNumber(newRx, factor, leadingZero);
+            node.attributes.ry = stringifyNumber(newRy, factor, leadingZero);
+          }
+          delete node.attributes.transform;
+        } else if (node.name == 'image') {
+          if (!isTranslation) return;
+
+          const x = Number(node.attributes.x || '0');
+          const y = Number(node.attributes.y || '0');
+          const corner = transformAbsolutePoint(matrix.data, x, y);
+
+          node.attributes.x = stringifyNumber(corner[0], factor, leadingZero);
+          node.attributes.y = stringifyNumber(corner[1], factor, leadingZero);
+          delete node.attributes.transform;
+        }
+      },
+    },
+  };
+};
+
+/**
+ * @param {Matrix} matrix
+ * @param {number} x
+ * @param {number} y
+ */
+const transformAbsolutePoint = (matrix, x, y) => {
+  const newX = matrix[0] * x + matrix[2] * y + matrix[4];
+  const newY = matrix[1] * x + matrix[3] * y + matrix[5];
+  return [newX, newY];
+};
+
+/**
+ * @param {Matrix} matrix
+ * @param {number} w
+ * @param {number} h
+ */
+const transformSize = (matrix, w, h) => {
+  const newW = matrix[0] * w + matrix[1] * h;
+  const newH = matrix[2] * w + matrix[3] * h;
+  return [Math.abs(newW), Math.abs(newH)];
+};
+
+/**
+ * @param {number} number
+ * @param {number} factor
+ * @param {boolean} leadingZero
+ */
+const stringifyNumber = (number, factor, leadingZero) => {
+  const rounded = Math.round(number * factor) / factor;
+  return leadingZero ? removeLeadingZero(rounded) : rounded.toString();
```

Comment:
It's confusing to have a param like `leadingZero: true` that do the opposite of what the name suggests. Can you name the param something like `removeLeadingZeros` instead so it's clearer?

Reply:
I did this for consistency with the name of the parameter that's used throughout plugins. Should I change it to `removeLeadingZeros` anyway?

Reply:
Hmm, good question. You're right, let's leave this one as is.

I didn't check the other plugins and forgot we had parameters for this already in other plugins. Since this is exposed, let's keep it consistent.

Thanks for raising that.